File: java/src/main/java/com/hachicore/demo/jooq/film/FilmService.java
package com.hachicore.demo.jooq.film;

import com.hachicore.demo.jooq.web.response.FilmWithActorPagedResponse;
import com.hachicore.demo.jooq.web.response.PagedResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FilmService {

    private final FilmRepository filmRepository;

    public FilmWithActorPagedResponse getFilmActorPageResponse(Long page, Long pageSize) {
        return new FilmWithActorPagedResponse(
                new PagedResponse(page, pageSize),
                filmRepository.findFilmWithActorsList(page, pageSize)
        );
    }

}

File: kotlin/src/main/kotlin/com/hachicore/demo/jooq/film/FilmService.kt
package com.hachicore.demo.jooq.film

import com.hachicore.demo.jooq.web.response.FilmWithActorPagedResponse
import com.hachicore.demo.jooq.web.response.PagedResponse
import org.springframework.stereotype.Service

@Service
class FilmService(
    val filmRepository: FilmRepository
) {

    fun getFilmActorPageResponse(page: Long, pageSize: Long): FilmWithActorPagedResponse {
        return FilmWithActorPagedResponse.of(
            PagedResponse(page, pageSize),
            filmRepository.findFilmWithActorList(page, pageSize)
        )
    }

}